src/services/aws/aws.ts

```diff
@@ -117,13 +117,14 @@ export const aws = createApi({
     }),
     endowment: builder.query<
       Endowment,
-      { id: number; fields?: (keyof Endowment)[] }
+      { id: number | string; fields?: (keyof Endowment)[] }
```

Comment:
This setup allows a small chance that id would be passed as a string by accident, but treated like a slug (`useEndowomentQuery("1")`), which would result in unintended behavior (although valid, it would return "endowment not found").

Suggestion: accept *either* `id` or `slug` _explicitly_:
```ts
    endowment: builder.query<
      Endowment,
      ({ id: number; slug?: never } | { id?: never; slug: string }) & {
        fields?: (keyof Endowment)[];
      }
    >({
      providesTags: ["profile"],
      query: ({ fields, id, slug }) => ({
        url: !!id ? `v7/endowments/${id}` : "v7/endowments", // might even write just `v7/endowments/${id}`
        params: {
          env: apiEnv,
          slug, // if `id` is passed, then this is undefined and will not be set
          ...(fields ? { fields: fields.join(",") } : {}),
        },
      }),
    }),
```
and update `useEndowment` wrapper to something similar:
```ts
export function useEndowment<T extends K[]>(
  idOrSlug: { id: number; slug?: never } | { id?: never; slug: string },
  fields?: T,
  options?: { skip: boolean }
) {
  const query = useEndowmentQuery(
    {
      ...idOrSlug, // pass on whatever was set
      fields,
    },
    options
  );

  return query as QueryState<
    T extends K[] ? Pick<Endowment, ArrayValues<T>> : Endowment
  >;
}
```

Reply:
fixed: https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2797/commits/3284a74febdbda19c9885f2a39e416bfa93985fd